### ISSUE TITLE
[FEA] Deprecate Series.as_mask

### ISSUE THREAD

**Issue by bdice** [rapidsai/cudf#9828]
**Is your feature request related to a problem? Please describe.**
Doctests (#9815) show the following doctest failing:
https://github.com/rapidsai/cudf/blob/582cc6e466c7d941e1b34893fd56fbd42fe90d68/python/cudf/cudf/core/series.py#L1783-L1793

**Describe the solution you'd like**
From a preliminary discussion with @vyasr, we probably want to deprecate and remove `as_mask` from the public API.

In the short term, I am removing the doctests for `Series.as_mask().to_host_array()` in #9815 (f881890a2da5e139358e3d2f531d663853b53748) because it's somewhat untestable and not helpful to demonstrate to users if we deprecate and remove the feature. It appears that the buffer has a size of 64 bytes. I assume that's the smallest allocation RMM can give, but that's not obvious to users who look at the output data of the doctest. Furthermore, the doctest shows uninitialized (garbage) data in the output of `to_host_array()` which is misleading.

**Comment by vyasr** [comment]
I think along with this we should think about what mask-related APIs we really should be exposing at the Python layer. For instance, should the `null_mask` or `nullable` properties even be accessible, or are they implementation details? `Series.set_mask` is already deprecated. Are there any public APIs where a bitmask is a reasonable expectation, or should all public APIs involving masks operate on boolean masks even if that is less performant (that's my opinion, IMO all bitmask operations should be exclusively internal, but perhaps there's a strong performance need to expose those).

**Comment by bdice** [comment]
I agree that many (all?) properties dealing with bitmasks should be considered implementation details for the Python API. I am investigating the internal/external usage of APIs like `Series.nullmask`, `Series.nullable`, `Series.as_mask`, and `Series.set_mask`.

**Comment by vyasr** [comment]
@shwina are you aware of any issues with proceeding here? @bdice's [last comment](https://github.com/rapidsai/cudf/issues/9828#issuecomment-999121929) shows what APIs still need to be updated.

**Comment by vyasr** [comment]
The comment above is still accurate. I think we can go ahead and remove these.
